File: files/pspro_lin64/examples/fortran/ShowDetail.f
      program ShowDetail

      include 'psprolib.inc'

!/******************************************************************
!*
!* This example demonstrates the sgShowServerDetail() function.
!* It retrieves information from the license server about all 
!* components that are checked out and by whom.  Plus some other 
!* interesting information for immediate display purposes.
!*
!******************************************************************/
      integer retVal,thread_id,sockfd,pid,count
      character*1024 msg
      character*256 process
      character*256 host
      character*64 user
      character*16 ip
      character*16 component
      character*12 version

      retVal = sgSetAttrInt(SG_ATTR_LICENSETYPE,SG_LICENSETYPE_FLOATING)

      retVal = sgConnect()
      if (retVal .ne. SG_SUCCESS) then
        retVal = sgGetLastErrorString(msg)
        write(*,'(A)') msg(:charlen(msg))
        stop
      end if
     
      do
        retVal = sgShowServerDetail(thread_id,sockfd,ip,pid,process,    &
     &                              user,host,component,version,count)
        if (retVal .eq. SG_NO_MORE_SERVER_DETAIL) goto 10
     
        write(*,'(I6,A,I5,3A,I5,11A,I4)')thread_id,' ',                 &
     &                               sockfd,' ',                        &
     &                               ip(:charlen(ip)),' ',              &
     &                               pid,' ',                           &
     &                               process(:charlen(process)),' ',    &
     &                               user(:charlen(user)),' ',          &
     &                               host(:charlen(host)),' ',          &
     &                               component(:charlen(component)),' ',&
     &                               version(:charlen(version)),' ',    &
     &                               count
      end do
  10  continue
    
      retVal = sgDisconnect()
      end program

File: files/pspro_lin64/examples/c/ShowDetail.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

#include "../../h/psprolib.h"

/******************************************************************
*
* This example demonstrates the sgShowServerDetail() function.
* It retrieves information from the license server about all 
* components that are checked out and by whom.  Plus some other 
* interesting information for immediate display purposes.
*
******************************************************************/
int main()
{
 int ret,thread_id,sockfd,pid,count;
 char msg[512],ip[16],process[256],user[64],host[256],component[33],version[12];

 sgSetAttrInt(SG_ATTR_LICENSETYPE,SG_LICENSETYPE_FLOATING);

 ret = sgConnect();
 if (ret != SG_SUCCESS) {
   sgGetLastErrorString(msg);
   printf("%s\n",msg);
   exit(0);
 }

 for (;;) {
     ret = sgShowServerDetail(&thread_id,&sockfd,ip,&pid,process,user,host,
                              component,version,&count);
     if (ret == SG_NO_MORE_SERVER_DETAIL) break;

     printf("(%d), %d, %d, %s, %d, %s, %s, %s, %s, %s, %d\n",ret,thread_id,sockfd,ip,pid,
			  process,user,host,component,version,count);
 }

 sgDisconnect();

exit(0);
}

